app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class DishesController < ApplicationController
+    def show
+        @dish = Dish.find(params[:id])
+    end
+
+    def add_ingredient
+        @dish = Dish.find(params[:id])
+        ingredient = Ingredient.find(params[:ingredient_id])
+        @dish.ingredients << ingredient
+        
+        redirect_to dish_path(@dish)
+    end
```

Comment:
Like I mentioned in your routes.rb file, this code wouldn't actually be in this controller. This work should be in a joins controller. So something like `DishIngredientsController` or `Dish::IngredientsController`.